src/nat/cli/commands/red_teaming/red_teaming_utils.py

```
@@ -0,0 +1,73 @@
+
+    with open(config_file, encoding='utf-8') as f:
+        if config_file.suffix in ('.yml', '.yaml'):
+            config_data = yaml.safe_load(f)
+        elif config_file.suffix == '.json':
+            config_data = json.load(f)
+        else:
+            raise ValueError(f"Unsupported file format: {config_file.suffix}. "
+                             "Use .yml, .yaml, or .json")
+
+    if not isinstance(config_data, dict):
+        raise ValueError(f"Red teaming config file must contain a dictionary, got {type(config_data)}")
```

Ruff (0.14.10)

Prefer `TypeError` exception for invalid type

(TRY004)